File: arrays_and_hashing/twoSums.py
from typing import List

class Solution:
  def twoSum(self, nums: List[int], target: int) -> List[int]:
    rest = {}
    for i in range(len(nums)):
      if nums[i] in rest.keys():
        return [rest[nums[i]], i]
      
      rest[target - nums[i]] = i
    return []

File: arrays_and_hashing/twoSums.java
import java.util.HashMap;

class Solution {
  public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> rest = new HashMap<>();
    for(int i = 0; i < nums.length; i++) {
      if(rest.containsKey(nums[i]))
        return new int[]{rest.get(nums[i]), i};

      rest.put(target - nums[i], i);
    }
    return new int[]{};
  }
}